fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraVersioning.java

```diff
@@ -185,29 +186,40 @@ public Response addVersion(@HeaderParam(MEMENTO_DATETIME_HEADER) final String da
 
             final boolean createFromExisting = isBlank(datetimeHeader);
 
-            if (requestContentType == null && !createFromExisting) {
-                throw new ClientErrorException("Content Type is required for creating a binary memento",
-                        UNSUPPORTED_MEDIA_TYPE);
-            }
-
             try {
                 LOGGER.info("Request to add version for date '{}' for '{}'", datetimeHeader, externalPath);
 
                 // Create memento
-                final FedoraResource memento;
-                if (resource instanceof FedoraBinary) {
-                    memento = versionService.createBinaryVersion(session.getFedoraSession(), resource(),
-                            mementoInstant, null, null, null, null);
-                } else {
+                FedoraResource memento = null;
+                final boolean isBinary = resource instanceof FedoraBinary;
+                if (isBinary) {
+                    final FedoraBinary binaryResource = (FedoraBinary) resource;
+                    if (createFromExisting) {
+                        memento = versionService.createBinaryVersion(session.getFedoraSession(),
+                                binaryResource, mementoInstant, storagePolicyDecisionPoint);
+                    } else {
+                        memento = createBinaryMementoFromRequest(binaryResource, mementoInstant,
+                                requestBodyStream, digest);
+                    }
+                }
+                // Create rdf memento if the request resource was an rdf resource or a binary from the
+                // current version of the original resource.
+                if (!isBinary || (isBinary && createFromExisting)) {
```

Comment:
Can remove `isBinary` from `(isBinary && createFromExisting)`... since it will always be `true` given the failure of the first condition.